joint_trajectory_controller/doc/userdoc.rst

```diff
@@ -101,6 +113,105 @@ A yaml file for using it could be:
               goal: 0.03
 
 
+.. _ROS 2 interface:
+
+ROS 2 interface of the controller [#f1]_
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+There are two mechanisms for sending trajectories to the controller:
+
+~/follow_joint_trajectory (action server) [control_msgs::action::FollowJointTrajectory]
+  Action server for commanding the controller.
+
+~/joint_trajectory (input topic) [trajectory_msgs::msg::JointTrajectory]
+  Topic for commanding the controller.
+
+Both use the ``trajectory_msgs/JointTrajectory`` message to specify trajectories, and require specifying values for all the controller joints (as opposed to only a subset) if ``allow_partial_joints_goal`` is not set to ``True``.
+
+The primary way to send trajectories is through the action interface, and should be favored when execution monitoring is desired.
+Action goals allow to specify not only the trajectory to execute, but also (optionally) path and goal tolerances.
+When no tolerances are specified, the defaults given in the parameter interface are used (see :ref:`parameters`).
+If tolerances are violated during trajectory execution, the action goal is aborted, the client is notified, and the current position is held.
+
+The topic interface is a fire-and-forget alternative. Use this interface if you don't care about execution monitoring.
+The controller's path and goal tolerance specification is not used in this case, as there is no mechanism to notify the sender about tolerance violations.
+Note that although some degree of monitoring is available through the ``~/query_state`` service and ``~/state`` topic it is much more cumbersome to realize than with the action interface.
+
+Further interfaces are:
+
+~/state (output topic) [control_msgs::msg::JointTrajectoryControllerState]
+  Topic publishing internal states with the update-rate of the controller manager.
+
+~/query_state (service) [control_msgs::srv::QueryTrajectoryState]
+  Query controller state at any future time.
+
+Preemption policy [#f1]_
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Only one action goal can be active at any moment, or none if the topic interface is used. Path and goal tolerances are checked only for the trajectory segments of the active goal.
+
+When an active action goal is preempted by another command coming from the action interface, the goal is canceled and the client is notified.
+
+Sending an empty trajectory message from the topic interface (not the action interface) will override the current action goal and not abort the action.
```

Comment:
I think this changed from ROS 1, is the described behavior correct (and intentional?)

Reply:
See https://github.com/ros-controls/ros2_controllers/blob/fa7499f0eb28ccca11408e703b8d5fce4f4deb30/joint_trajectory_controller/src/joint_trajectory_controller.cpp#L959-L964

Reply:
I think it will be overridden, but it should not be. The explanation is correct, but I  don't think it is desirable.

Reply:
Ok, so this is fine for this PR -> I created a new issue based on your comment.

Reply:
https://github.com/ros-controls/ros2_controllers/issues/573